experimentation/tools/sorald/handled_rules.py

```diff
@@ -0,0 +1,203 @@
+import argparse
+import dataclasses
+import json
+import pathlib
+import re
+import subprocess
+import sys
+
+import jinja2
+
+from os.path import dirname, abspath, join
+from typing import Dict, List, Union
+
+from sorald._helpers import sonar_metadata, jsonkeys
+
+ENCODING = "utf8"
+
+BUG = []
+BUG_DETAIL = []
+
+CODE_SMELL = []
+CODE_SMELL_DETAIL = []
+
+VULNERABILITY = []
+VULNERABILITY_DETAIL = []
```

Comment:
This can be combined to form two lists - `List` and `Detail`. Both would render for all repairs for all violation types. But that can be an enhancement for another PR.

Reply:
In general, mutable globals is a one-way ticket to bug town. For example, what happens if you import this script and run it twice? It also suggests that there's a function somewhere with a side effect of mutating these globals, which is also a bad smell.

I'd make these a return value of some function instead and use that where appropriate.

Reply:
Yeah, I was thinking about this too. I initially thought I would write `handled_rules.py` as a class so that I can contain such private variables and then return when classification is over.